cpp_test_ds/fwd_ds/CMakeLists.txt

```diff
@@ -5,5 +5,10 @@ set(SOURCES ClassFactory.cpp
             main.cpp)
 
 add_executable(FwdTest ${SOURCES})
-target_include_directories(FwdTest PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
-target_link_libraries(FwdTest PUBLIC tango ${CMAKE_DL_LIBS})
+if(CMAKE_VERSION VERSION_GREATER 2.8.12)
+    target_include_directories(FwdTest PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+    target_link_libraries(FwdTest PUBLIC tango ${CMAKE_DL_LIBS})
+else()
+    set_property(TARGET FwdTest APPEND PROPERTY INCLUDE_DIRECTORIES ${CMAKE_CURRENT_SOURCE_DIR})
+    target_link_libraries(FwdTest LINK_PUBLIC tango ${CMAKE_DL_LIBS})
```

Comment:
In target_link_libraries, we could actually use LINK_PUBLIC instead of PUBLIC all the time since this seems to be supported in recent CMake versions in order to be compatible with older CMake versions. We could always change it later on if this becomes deprecated in future CMaked versions.